builtin/logical/pki/backend.go

```diff
@@ -233,3 +236,14 @@ func (b *backend) metricsWrap(callType string, roleMode int, ofunc roleOperation
 		return resp, err
 	}
 }
+
+// initialize is used to perform a possible PKI storage migration if needed
+func (b *backend) initialize(ctx context.Context, req *logical.InitializationRequest) error {
+	// Only primary should perform the upgrade.
+	if b.System().ReplicationState().HasState(consts.ReplicationPerformanceSecondary | consts.ReplicationPerformanceStandby | consts.ReplicationDRSecondary) {
+		return nil
+	}
+
+	logger := b.Logger().Named("initialize")
```

Comment:
Can it ever be that everyone thinks they're a PerfSecondary/PerfStandby/DRSecondary during initialization because the last leader wasn't brought up, perhaps with an auto-unseal mechanism? Or is that unlikely?

Reply:
It's a common pattern I see in Vault so I don't believe so but I can't point to proof of that at the moment. One thing you made me think of that is missing in this statement though is local mounts...

Reply:
After the morning coffee, no there isn't a way that a cluster/node can be switched state during initialization of it's cluster type based on leader election. Technically a primary cluster could be converted to a secondary but it will get wiped out and reset anyways.